DoLah/Models/Calendar.cpp

```diff
@@ -158,6 +158,30 @@ namespace DoLah {
         this->searchedList = results;
     }
 
+    void Calendar::searchDate(std::tm from, std::tm to) {
+        std::vector<AbstractTask*> results;
+
+        for (int i = 0; i < taskList.size(); i++) {
+            std::vector<std::tm> dates = getDates(taskList[i]);
+            if (dates.size() == 0) {
+                continue;
+            }
+            if (TimeManager::compareTime(from, dates[0]) >= 0 && TimeManager::compareTime(dates[0], to) >= 0) {
+                results.push_back(taskList.at(i));
+            }
```

Comment:
No it asserts that ```from <= deadline or endDate <= to```.
Cannot understand your remark on the past and future. Any simple example?